internal/venice-common/src/main/java/com/linkedin/venice/meta/Partition.java

```diff
@@ -30,10 +31,52 @@ public class Partition {
   private final int id;
 
   private final Map<String, List<Instance>> stateToInstancesMap;
+  private final Map<HelixState, List<Instance>> helixStateToInstancesMap;
```

Comment:
Looks like we can get aways with just storing the Instance::getNodeID instead of the full instance objects. but savings probably not worth the refactoring.

Reply:
Hmm... perhaps, but this `List<Instance>` ends up being used in a ton of places, so the cost of refactoring seems steep (if it is possible at all, not sure).

In terms of overhead, IDK if it changes anything. Either way it'll be a list of pointers to objects on the heap. Whether the pointers point to Strings or `Instance` objects doesn't matter, assuming those objects are going to be on the heap no matter what, due to being referenced from elsewhere.

Reply:
yeah, Instances are already pre-allocated before Partition class creates List of those.